src/legacy/legacy.c

```diff
@@ -0,0 +1,57 @@
+/*PGR-GNU*****************************************************************
+File: legacy.c
+
+Copyright (c) 2017 pgRouting developers
+Mail: project@pgrouting.org
+
+Copyright (c) 2018-2020 Regina Obe <lr@pcorp.us>
+------
+
+This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; either version 2 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program; if not, write to the Free Software
+Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+ ********************************************************************PGR-GNU*/
+
+/******************************************************************************
+ * This file is to hold functions we no longer use,
+ * but we need to keep because they were used at one time behind SQL API functions.
+ * This is to ease pg_upgrade upgrades
+ *
+ * All functions in this file should throw an error telling the user to upgrade
+ * the install
+ *
+ *****************************************************************************/
+
+#include "postgres.h"
+#include "utils/builtins.h"
+
+#define PGROUTING_DEPRECATE(version, funcname) \
+    Datum funcname(PG_FUNCTION_ARGS); \
+    PG_FUNCTION_INFO_V1(funcname); \
+    Datum funcname([[maybe_unused]] PG_FUNCTION_ARGS) \
```

Comment:
@cvvergara I couldn't reproduce macos CI error on my macOS 15.2 environment, but macOS 14.7.2 in CI seems not to support `[[maybe_unused]]` attribute in C++17.
https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2016/p0212r1.pdf

From the following article, it seems to be better to use `__attribute__((unused))` instead of `[[maybe_unused]]`.
https://stackoverflow.com/questions/31909631/c11-style-unused-attribute-in-gcc